packages/core/src/entities/feature.ts

```diff
@@ -32,24 +32,26 @@ export class RuntimeFeature<
         public dependencies: RunningFeatures<Deps, string>
     ) {}
 
-    public addRunHandler(fn: () => void, envName: string) {
+    public addRunHandler(fn: () => unknown, envName: string) {
         this.runHandlers.add(envName, fn);
     }
     public addOnDisposeHandler(fn: DisposeFunction, envName: string) {
         this.disposeHandlers.add(envName, fn);
     }
-    public [RUN](context: RuntimeEngine, envName: string) {
+    public async [RUN](context: RuntimeEngine, envName: string): Promise<void> {
         if (this.running) {
             return;
         }
         this.running = true;
+        const runPromises: Array<unknown> = [];
         for (const dep of this.feature.dependencies) {
-            context.runFeature(dep, envName);
+            runPromises.push(context.runFeature(dep, envName));
         }
         const envRunHandlers = this.runHandlers.get(envName) || [];
         for (const handler of envRunHandlers) {
-            handler();
+            runPromises.push(handler());
         }
+        await Promise.all(runPromises)
```

Comment:
maybe better to return this, so that consumer could decide weather to await for that or not

Reply:
it's the same. think about it. the only reason to return is if u want to propagate the promised value (not the case).